src/external/SMIOL/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+
+find_package(MPI REQUIRED COMPONENTS C Fortran)
+find_package(NetCDF REQUIRED COMPONENTS Fortran C)
+find_package(PnetCDF REQUIRED COMPONENTS Fortran C)
+
+# Specify the library source files
+set(SMIOL_C_SOURCES smiol.c smiol_utils.c)
+set(SMIOL_F_SOURCES smiolf.F90)
+
+# Create the C library
+add_library(smiol ${SMIOL_C_SOURCES})
+add_library(${PROJECT_NAME}::external::smiol ALIAS smiol)
+target_compile_definitions(smiol PRIVATE SMIOL_PNETCDF SINGLE_PRECISION)
+
+# Create the Fortran library
+add_library(smiolf ${SMIOL_F_SOURCES})
+enable_language(Fortran)
+mpas_fortran_target(smiolf)
+add_library(${PROJECT_NAME}::external::smiolf ALIAS smiolf)
+target_compile_definitions(smiolf PRIVATE SMIOL_PNETCDF )
```

Comment:
When compiling this branch on my local machine, the build fails because the `smiol` target cannot find `mpi.h`. I attached the build log for reference.
[log.make.txt](https://github.com/user-attachments/files/18213005/log.make.txt)
 The issue is resolved by adding `target_include_directories(smiol PRIVATE ${MPI_INCLUDE_PATH})` after `target_compile_definitions(smiol PRIVATE SMIOL_PNETCDF SINGLE_PRECISION)`.

Reply:
Thanks for testing and finding this.